src/wallet/asyncrpcoperation_saplingmigration.cpp

```diff
@@ -185,7 +185,7 @@ libzcash::SaplingPaymentAddress AsyncRPCOperation_saplingmigration::getMigration
         pwalletMain->GetSaplingFullViewingKey(ivk, fvk) &&
         pwalletMain->HaveSaplingSpendingKey(fvk))) {
         // Sapling account 0 must be the first address returned by GenerateNewSaplingZKey
-        assert(pwalletMain->GenerateNewSaplingZKey() == toAddress);
+        assert(pwalletMain->GenerateNewSaplingZKey(false) == toAddress);
```

Comment:
AFAICT this means that if the user has never generated a Sapling address, the migrated funds will disappear from their wallet (e.g. `z_gettotalbalance` will not show them), because this call previously had a side-effect of adding the default address into the wallet. If that side-effect is being removed here, at what point _will_ the default address be added to the wallet?

Reply:
I added this to make it easier to test. I was also unsure whether or not it was correct to have this side-effect here. Looking at it again. I think it is preferable to leave this in. I will remove the commit with this refactoring.